dask/array/tests/test_slicing.py

```diff
@@ -421,6 +421,13 @@ def test_slicing_and_chunks():
     assert t.chunks == ((8, 8), (6, 6))
 
 
+def test_slicing_and_unknown_chunks():
+    a = da.ones((10, 5), chunks=5)
+    a._chunks = ((np.nan, np.nan), (5,))
```

Comment:
I think @ian-r-rose suggested coming to empty chinks naturally rather than removing them like this. Something like:

```suggestion
    a = a[a>20]
```

Reply:
I forgot to post why I didn't do this. 

This example doesn't seem to result in unknown chunks, that trigger the issue we are trying to resolve. 
I tried this and it had no problems. There are certain situations where we have nan in the chunks but we have ways of solving this. I'm not sure what's the right example that triggers the type of unknown chunks that causes this problem.
 
 It seems we do this overwriting of the chinks in other places, see line 935 of this bit https://github.com/dask/dask/blob/238b5dbfb1bd42a86d63df4fe958037f70e8f87b/dask/array/tests/test_slicing.py#L919-L939 .  In fact I was wondering if we should remove the xfail of this case

Reply:
ah ok. It's certainly not a big deal if it doesn't just work. Thanks for the explanation.